src/applications/ivc-champva/10-7959f-1/helpers/maskSSN.jsx

```diff
@@ -0,0 +1,57 @@
+import { formatSSN } from 'platform/utilities/ui';
+import React, { useState, useEffect, useRef } from 'react';
+import vaTextInputFieldMapping from 'platform/forms-system/src/js/web-component-fields/vaTextInputFieldMapping';
+import { VaTextInput } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+function maskSSN(ssnString = '') {
+  const strippedSSN = ssnString.replace(/[- ]/g, '');
+  const maskedSSN = strippedSSN.replace(/^\d{1,5}/, digit =>
+    digit.replace(/\d/g, '●'),
+  );
+  return formatSSN(maskedSSN);
+}
+
+/**
+ * @param {WebComponentFieldProps} props */
+export default function HandlePrefilledSSN(fieldProps) {
+  const props = vaTextInputFieldMapping(fieldProps);
+  const [val, setVal] = useState(props.value);
+  const [displayVal, setDisplayVal] = useState(props.value);
+  const vaTextInput = useRef();
+
+  useEffect(() => {
+    setVal(maskSSN(val));
```

Comment:
We need to not set the actual val to be the masked SSN, because then we lose the first few digits and the form throws an error when navigating to the next page
![Screenshot 2024-06-11 at 08 22 59](https://github.com/department-of-veterans-affairs/vets-website/assets/18408628/d08a02d5-5a39-4f88-a702-b4de635e708a)
![Screenshot 2024-06-11 at 08 23 08](https://github.com/department-of-veterans-affairs/vets-website/assets/18408628/58de68e5-cfc5-45cb-a06c-7f3732ab9152)

Reply:
So this seems to be working now. It does show the masked value in the console, but I checked and it's getting all the numbers as a real value now, so it SHOULD be good.